### PR TITLE
Initial changes for Mode

### LINKED ISSUE #1542: document primitive `mode` arg
followup from https://github.com/Qiskit/documentation/issues/1486#issuecomment-2160933818

```
with Batch(backend=backend):
    sampler = SamplerV2(mode=batch)
```

### PR DESCRIPTION
**Pull request by beckykd**
closes #1542 

Frank did a lot of the code changes already (#1580 ), but the migration guides weren't included, so I still need to update that code.

### CODE REVIEW

File: docs/api/migration-guides/qiskit-runtime-examples.mdx [hunk 1]
```diff
@@ -225,7 +225,7 @@ Output
...
-The legacy workflow required additional elements and steps to compute an expectation value. The `QuantumInstance` class stored a `PassManager` and a `Backend` instance, and wrapped the conversion step to ISA circuits and `backend.run` calls.  The `CircuitSampler` class from `qiskit.opflow` wrapped the `QuantumInstance` `run` and `transpile` methods. This degree of nesting is no longer supported, the new workflow allows to access and directly manipulate all the key components of the computation (backend, pass manager, circuits and observables) at all stages of the algorithm. 
+The legacy workflow required additional elements and steps to compute an expectation value. The `QuantumInstance` class stored a `PassManager` and a `Backend` instance, and wrapped the conversion step to ISA circuits and `backend.run` calls.  The `CircuitSampler` class from `qiskit.opflow` wrapped the `QuantumInstance` `run` and `transpile` methods. This degree of nesting is no longer supported, the new workflow allows to access and directly manipulate all the key components of the computation (backend, pass manager, circuits and observables) at all stages of the algorithm.
```
Comment (abbycross) [inline docs/api/migration-guides/qiskit-runtime-examples.mdx:228]:
```suggestion
The legacy workflow required additional elements and steps to compute an expectation value. The `QuantumInstance` class stored a `PassManager` and a `Backend` instance, and wrapped the conversion step to ISA circuits and `backend.run` calls.  The `CircuitSampler` class from `qiskit.opflow` wrapped the `QuantumInstance`, `run`, and `transpile` methods. This degree of nesting is no longer supported; the new workflow allows to access and directly manipulate all the key components of the computation (backend, pass manager, circuits and observables) at all stages of the algorithm.

```

**Review by abbycross — APPROVED** [review]
👍

File: docs/api/migration-guides/v2-primitives.mdx [hunk 2]
```diff
@@ -288,7 +288,7 @@ backend = service.least_busy(operational=True, simulator=False)
 options = Options()
 # This uses auto complete.
 options.resilience_level = 2
-estimator = Estimator(backend=backend, options=options)
+estimator = Estimator(mode=backend, options=options)
```
Comment (jyu00) [inline docs/api/migration-guides/v2-primitives.mdx:291]: We didn't bother making the change for V1 primitives 😛

File: docs/guides/execution-modes.mdx [hunk 3]
```diff
@@ -7,11 +7,11 @@ description: An overview of the available execution modes in Qiskit Runtime; ses
 
 There are several ways to run workloads, depending on your needs.
 
-**Job mode**: A single primitive request of the estimator or the sampler made without a context manager. Circuits and inputs are packaged as primitive unified blocs (PUBs) and submitted as an execution task on the quantum computer.
+**Job mode**: A single primitive request of the estimator or the sampler made without a context manager. Circuits and inputs are packaged as primitive unified blocs (PUBs) and submitted as an execution task on the quantum computer. To run in job mode, specify `mode=backend` or pass the backend name when instantiating a primitive. See [Primitives examples](primitives-examples) for examples.
```
Comment (jyu00) [inline docs/guides/execution-modes.mdx:10]: We actually also deprecated passing the backend name when initializing a primitive. If you do `SamplerV2(mode="ibm_osaka")`, for example, it actually uses the default account under the cover, which could be unexpected to users. 

```suggestion
**Job mode**: A single primitive request of the estimator or the sampler made without a context manager. Circuits and inputs are packaged as primitive unified blocs (PUBs) and submitted as an execution task on the quantum computer. To run in job mode, specify `mode=backend` when instantiating a primitive. See [Primitives examples](primitives-examples) for examples.
```

File: docs/guides/get-started-with-primitives.mdx [hunk 4]
```diff
@@ -73,7 +73,7 @@ isa_observable = observable.apply_layout(isa_circuit.layout)
 
 ### 3. Initialize Qiskit Runtime Estimator
 
-When you initialize the Estimator, pass in the system or simulator you previously selected as the target.
+When you initialize the `Estimator`, use the `mode` parameter to specify the mode you want it to run in.  Possible values are `batch`, `session`, or `backend` for job execution mode. For more information, see [Introduction to Qiskit Runtime execution modes.](execution-modes)
```
Comment (jyu00) [inline docs/guides/get-started-with-primitives.mdx:76]: I'd like to point out `batch` etc are name of the objects, not literally "batch". 

```suggestion
When you initialize the `Estimator`, use the `mode` parameter to specify the mode you want it to run in.  Possible values are `batch`, `session`, or `backend` objects for batch, session, and job execution mode, respectively.. For more information, see [Introduction to Qiskit Runtime execution modes.](execution-modes)
```

File: docs/guides/get-started-with-primitives.mdx [hunk 5]
```diff
@@ -188,7 +188,7 @@ isa_circuit = pm.run(circuit)
 
 ### 3. Initialize the Qiskit Runtime Sampler
 
-When you initialize the `Sampler`, pass in the simulator or system you previously selected as the target.
+When you initialize the `Sampler`, use the `mode` parameter to specify the mode you want it to run in.  Possible values are `batch`, `session`, or `backend` for job execution mode. For more information, see [Introduction to Qiskit Runtime execution modes.](execution-modes)
```
Comment (jyu00) [inline docs/guides/get-started-with-primitives.mdx:191]: Same as above.

File: docs/guides/execution-modes.mdx [hunk 6]
```diff
@@ -7,11 +7,15 @@ description: An overview of the available execution modes in Qiskit Runtime; ses
 
 There are several ways to run workloads, depending on your needs.
 
-**Job mode**: A single primitive request of the estimator or the sampler made without a context manager. Circuits and inputs are packaged as primitive unified blocs (PUBs) and submitted as an execution task on the quantum computer.
+<Admonition type="note">
+The `mode` option is only available for V2 primitives.
```
Comment (jyu00) [inline docs/guides/execution-modes.mdx:11]: do we want to also say it needs qiskit-ibm-runtime >= 0.24?

**Review by jyu00 — APPROVED** [review]

**Pull request merged**